worlds/khrecom/Rules.py

```diff
@@ -0,0 +1,466 @@
+        lambda state: state.has("World Card Atlantica", player))
+    add_rule(khrecomworld.get_location("Defeat 1 Heartless Bandit"),
+        lambda state: state.has("World Card Agrabah", player))
+    add_rule(khrecomworld.get_location("Defeat 2 Heartless Bandit"),
+        lambda state: state.has("World Card Agrabah", player))
+    add_rule(khrecomworld.get_location("Defeat 3 Heartless Bandit"),
+        lambda state: state.has("World Card Agrabah", player))
+    add_rule(khrecomworld.get_location("Defeat 1 Heartless Barrel Spider"),
+        lambda state: (
+            state.has("World Card Monstro", player)
+            or state.has("World Card Agrabah", player)
+            or state.has("World Card Neverland", player)
+        ))
+    add_rule(khrecomworld.get_location("Defeat 2 Heartless Barrel Spider"),
+        lambda state: (
+            state.has("World Card Monstro", player)
+            or state.has("World Card Agrabah", player)
+            or state.has("World Card Neverland", player)
+        ))
+    add_rule(khrecomworld.get_location("Defeat 3 Heartless Barrel Spider"),
+        lambda state: (
+            state.has("World Card Monstro", player)
+            or state.has("World Card Agrabah", player)
+            or state.has("World Card Neverland", player)
+        ))
+    add_rule(khrecomworld.get_location("Defeat 1 Heartless Bouncywild"),
+        lambda state: state.has("World Card Olympus Coliseum", player))
+    add_rule(khrecomworld.get_location("Defeat 2 Heartless Bouncywild"),
+        lambda state: state.has("World Card Olympus Coliseum", player))
+    add_rule(khrecomworld.get_location("Defeat 3 Heartless Bouncywild"),
+        lambda state: state.has("World Card Olympus Coliseum", player))
+    add_rule(khrecomworld.get_location("Defeat 1 Heartless Creeper Plant"),
+        lambda state: (
+            state.has("World Card Wonderland", player)
+            or state.has("World Card Halloween Town", player)
+            or state.has("World Card Destiny Islands", player)
+        ))
+    add_rule(khrecomworld.get_location("Defeat 2 Heartless Creeper Plant"),
+        lambda state: (
+            state.has("World Card Wonderland", player)
+            or state.has("World Card Halloween Town", player)
+            or state.has("World Card Destiny Islands", player)
+        ))
+    add_rule(khrecomworld.get_location("Defeat 3 Heartless Creeper Plant"),
+        lambda state: (
+            state.has("World Card Wonderland", player)
+            or state.has("World Card Halloween Town", player)
+            or state.has("World Card Destiny Islands", player)
+        ))
+    add_rule(khrecomworld.get_location("Defeat 1 Heartless Crescendo"),
+        lambda state: (
+            state.has("World Card Wonderland", player)
+            or state.has("World Card Neverland", player)
+            or state.has("World Card Destiny Islands", player)
+        ))
```

Comment:
```suggestion
    add_rule(khrecomworld.get_location("Defeat 1 Heartless Crescendo"),
        lambda state: state.has_any({"World Card Wonderland", "World Card Neverland", "World Card Destiny Islands"}, player))
```
`has_any` is probably more readable here and in other, similar rules here